controls/SV-244552.rb

```diff
@@ -1,56 +1,68 @@
+/etc/sysctl.d/*.conf
+/run/sysctl.d/*.conf
+/usr/local/lib/sysctl.d/*.conf
+/usr/lib/sysctl.d/*.conf
+/lib/sysctl.d/*.conf
+/etc/sysctl.conf'
+  desc 'check', 'Verify RHEL 8 does not accept IPv4 source-routed packets by default.
 
-    Check the value of the accept source route variable with the following
-command:
+Check the value of the accept source route variable with the following command:
 
-    $ sudo sysctl net.ipv4.conf.default.accept_source_route
+$ sudo sysctl net.ipv4.conf.default.accept_source_route
 
-    net.ipv4.conf.default.accept_source_route = 0
+net.ipv4.conf.default.accept_source_route = 0
 
-    If the returned line does not have a value of \"0\", a line is not
-returned, or the line is commented out, this is a finding.
-  "
-  desc  'fix', "
-    Configure RHEL 8 to not forward IPv4 source-routed packets by default with
-the following command:
+If the returned line does not have a value of "0", a line is not returned, or the line is commented out, this is a finding.
 
-    $ sudo sysctl -w net.ipv4.conf.default.accept_source_route=0
+Check that the configuration files are present to enable this network parameter.
 
-    If \"0\" is not the system's default value then add or update the following
-line in the appropriate file under \"/etc/sysctl.d\":
+$ sudo grep -r net.ipv4.conf.default.accept_source_route /run/sysctl.d/*.conf /usr/local/lib/sysctl.d/*.conf /usr/lib/sysctl.d/*.conf /lib/sysctl.d/*.conf /etc/sysctl.conf /etc/sysctl.d/*.conf
 
-    net.ipv4.conf.default.accept_source_route=0
-  "
+/etc/sysctl.d/99-sysctl.conf: net.ipv4.conf.default.accept_source_route = 0
+
+If "net.ipv4.conf.default.accept_source_route" is not set to "0", is missing or commented out, this is a finding.
+
+If conflicting results are returned, this is a finding.'
+  desc 'fix', 'Configure RHEL 8 to not forward IPv4 source-routed packets by default.
+
+Add or edit the following line in a system configuration file, in the "/etc/sysctl.d/" directory:
+
+net.ipv4.conf.default.accept_source_route=0
+
+Remove any configurations that conflict with the above from the following locations:
+/run/sysctl.d/*.conf
+/usr/local/lib/sysctl.d/*.conf
+/usr/lib/sysctl.d/*.conf
+/lib/sysctl.d/*.conf
+/etc/sysctl.conf
+/etc/sysctl.d/*.conf
+
+Load settings from all system configuration files with the following command:
+
+$ sudo sysctl --system'
   impact 0.5
   tag severity: 'medium'
   tag gtitle: 'SRG-OS-000480-GPOS-00227'
   tag gid: 'V-244552'
-  tag rid: 'SV-244552r743905_rule'
+  tag rid: 'SV-244552r858803_rule'
   tag stig_id: 'RHEL-08-040249'
-  tag fix_id: 'F-47784r743904_fix'
+  tag fix_id: 'F-47784r858802_fix'
   tag cci: ['CCI-000366']
   tag nist: ['CM-6 b']
 
   if virtualization.system.eql?('docker')
```

Comment:
Check the logic here, with the statement at the end about only applying to incoming traffic when acting as a router.